### ISSUE TITLE
confuse about lora type

### ISSUE THREAD

**Issue by yiranxie233** [McGill-NLP/diffusion-itm#6]
Sorry for bother you again.
When I eval the diff_itm, I used the checkpoint you provide, and noticed that there is lora type when use the lora_dir, what's difference between these lora types, and does lora type influence the retrieval result?

**Comment by BennoKrojer** [comment]
Hi,

Some of them are outdated (for example "relativistic") but they just referred to the exact finetuning method, i.e. how I used hard negatives. And they are only used for naming the folder for logging.

Are you still struggling in general to reproduce it? Unfortunately I don't have the bandwidth to dig into in detail right now. I vaguele remember reproducing it again myself when releasing the repo. 

You probably have to set the sampling steps to something like 100. It takes longer though then. Let me know if that works!

`python diffusion_itm.py --task flickr30k --seed 42 --subset --img_retrieval --lora_dir checkpoints/hardneg-checkpoint-5000 --sampling_steps 100`

Best,
Benno

**Comment by yiranxie233** [comment]
Hi Benno,
Thank you for your reply! 
I am still reproducing it and want to apply it to my field.
I used sampling_steps 50 and get better result than default. I looked at the code of the hardneg.py, and I don't know how to get result of NoNeg, I think the dataset is based on the negatives, can you give me some advise about how to get the NoNeg one?
Best,
Xie

**Comment by BennoKrojer** [comment]
Hi,

NoNeg just means vanilla finetuning with lora. So each image-text pair would actually fit together. So you would pass no argparse arguments that indicate negative training, and check in the code that it is indeed running normal training. Perhaps the easiest way to do this would be just to use the default script on which i based my code:

https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_lora.py

Good luck,
Benno

**Comment by yiranxie233** [comment]
Hi Benno,

Thank you for your reply.
I will try your advise later! Thanks a lot!!
Yesterday I reproduced the code with different lora dirs on flickr30k on another machine, after I ran code, I found the hardneg and without lora_dir get same results in Rank@1, the command  and result will show later, here I want to check something about prepare and hope you can give me some advise if I did something wrong. First, I used the sd2.1 base [here](https://huggingface.co/stabilityai/stable-diffusion-2-1-base/tree/main) , after that I faced the question same as #2 did not have the noise_samples.pkl, after solved this problem by making diffusers directory , then the code ran successfully. The flickr pictures I used the web you provide to download [here](https://www.kaggle.com/datasets/hsankesara/flickr-image-dataset).

Hope to get your reply!!
Best, 
XIe

![image](https://github.com/user-attachments/assets/09a8aa79-13b8-4e0c-bd61-71ed45a795c4)
![image-20241020164950545](https://github.com/user-attachments/assets/02ece313-d018-4132-834a-6b60e9a6cd87)
![image-20241020160811041](https://github.com/user-attachments/assets/56163612-a7fd-4971-8879-c404fa2ab5aa)

**Comment by yiranxie233** [comment]
Hi Benno,


I ran code today, and didn't use the hyper-parameter *subset*, I find that the result is similar to the paper report, although the speed were longer than add *subset*. I will show result in the end of comment.
Thank you for your reply patiently!! If I have other questions I will ask you again.

Best, 
Xie
![image](https://github.com/user-attachments/assets/e6bb9bd6-2876-4928-8454-8c1fa2ab8b11)

**Comment by BennoKrojer** [comment]
I am glad to hear that things are working slowly! Feel free to email what exactly your research question is btw, more out of curiosity :) I am not working in this direction anymore but it is still interesting what comes out of it

**Comment by yiranxie233** [comment]
Thank you so much!!!
